OutOfSchool/OutOfSchool.IdentityServer/ProfileService.cs

```diff
@@ -73,5 +81,30 @@ private async Task<string> GetPermissionsForUser(string userName, string roleNam
 
             return permissionsForUser;
         }
+
+        // Get subrole for user
+        private async Task<Subrole> GetSubroleByUserName(string userName)
+        {
+            var userToLogin = await userManager.FindByNameAsync(userName);
```

Comment:
your method here and `GetPermissionsForUser` both are finding the user by name, so probaly it's worth moving it out of both methods and passing user as args.

Reply:
Done